src/autolabel/few_shot/label_diversity_example_selector.py

```diff
@@ -0,0 +1,153 @@
+"""Example selector that selects examples based on SemanticSimilarity."""
+from __future__ import annotations
+
+from itertools import groupby
+from operator import itemgetter
+from typing import Any, Dict, List, Optional, Type
+
+from langchain.embeddings.base import Embeddings
+from langchain.prompts.example_selector.base import BaseExampleSelector
+from langchain.vectorstores.base import VectorStore
+from pydantic import BaseModel, Extra
+
+
+def sorted_values(values: Dict[str, str]) -> List[Any]:
+    """Return a list of values in dict sorted by key."""
+    return [values[val] for val in sorted(values)]
+
+
+class LabelDiversityRandomExampleSelector(BaseExampleSelector, BaseModel):
+    """Example selector that selects examples based on SemanticSimilarity."""
+
+    examples: List[dict]
+    """A list of the examples that the prompt template expects."""
+    k: int = 1
```

Comment:
"k" parameter in https://github.com/refuel-ai/autolabel/pull/436/files#diff-374f83319a2bfe5304bc20448393efad6a01c24c64d1cc276180d67141c99d65L61 is for the total number of few-shot examples. we need to map this to num examples per label by dividing it by number of labels?